vike/node/runtime/html/stream.ts

```diff
@@ -115,15 +115,17 @@ async function streamReadableNodeToString(readableNode: StreamReadableNode): Pro
 }
 
 async function streamReadableWebToString(readableWeb: ReadableStream): Promise<string> {
-  let str: string = ''
   const reader = readableWeb.getReader()
+  const decoder = new TextDecoder()
+  let str: string = ''
+
   while (true) {
     const { done, value } = await reader.read()
-    if (done) {
-      break
-    }
-    str += value
+    if (done) break
+    str += decoder.decode(value, { stream: true })
   }
+
+  str += decoder.decode()
```

Comment:
Curious: why this line?

Reply:
![image](https://github.com/user-attachments/assets/b1721542-1d26-4543-ab5f-dca7ee484d95)

Reply:
The final decoder.decode() is actually only necessary in a very specific scenario:
If the very last chunk of the stream ends with the beginning of a multi-byte character, that incomplete character data will be held by the decoder and not output. The final decoder.decode() call will then output a replacement character (usually �) for this incomplete sequence.